google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableResult.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.gax.paging.Page;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Iterators;
+import java.io.Serializable;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+public class TableResult implements Page<FieldValueList>, Serializable {
+
+  private static final long serialVersionUID = -4831062717210349819L;
+
+  @Nullable private final Schema schema;
+  private final long totalRows;
+  private final Page<FieldValueList> pageNoSchema;
+
+  TableResult(Schema schema, long totalRows, Page<FieldValueList> pageNoSchema) {
+    this.schema = schema;
+    this.totalRows = totalRows;
+    this.pageNoSchema = checkNotNull(pageNoSchema);
+  }
+
+  /** Returns the schema of the results. Null if the schema is not supplied. */
+  public Schema getSchema() {
+    return schema;
+  }
+
+  /**
+   * Returns the total number of rows in the complete result set, which can be more than the number
+   * of rows in the first page of results returned by {@link #getValues()}.
+   *
+   * <p>If this instance is a result of a dry-run query, returns {@code 0}.
```

Comment:
Would someone even want this object if running a dry-run query? We should probably point them at `Job.waitFor()`

Reply:
Let's remove this for now. I'm not even sure if `waitFor` will work for dry-runs.